test/test_network.h

```diff
@@ -233,7 +233,7 @@ TEST(network, train_predict) {
 
   net << fully_connected_layer(2, 10) << tanh_layer()
       << fully_connected_layer(10, 2) << tanh_layer();
-
+  net.init_parameters();
```

Comment:
@karandesai-96 so now is it mandatory to explicitly initialize the parameters? It should be automatic unless the user wants an specific initialization or import from whatever resource.

Reply:
@edgarriba Well I can call init_parameters of tail layer in `<<` operator. How about that ?

Reply:
@karandesai-96 sounds good